projects/libs/core-library-typescript/src/common-library/signatures.ts

```diff
@@ -0,0 +1,65 @@
+import * as crypto from "crypto";
+import * as stringify from "json-stable-stringify";
+import * as fs from "fs";
+import * as _ from "lodash";
+
+export interface SignaturesConfig {
+  publicKeysPath?: string;
+  privateKeyPath?: string;
+}
+
+export class Signatures {
+  config: SignaturesConfig;
+  SIGNATURE_FORMAT: crypto.HexBase64Latin1Encoding = "base64";
+  SIGNATURE_TYPE = "sha256";
+
+  private publicKeys = new Map<string, string>();
+  private privateKey: string;
+
+  public constructor(signaturesConfig: SignaturesConfig) {
+    this.config = signaturesConfig;
+
+    this.readPrivateKey();
+    this.readPublicKeys();
+  }
+
+  private readPrivateKey() {
+    if (this.config.privateKeyPath) {
+      this.privateKey = fs.readFileSync(this.config.privateKeyPath).toString();
+    }
+  }
+
+  private readPublicKeys() {
+    if (this.config.publicKeysPath) {
+      fs.readdirSync(this.config.publicKeysPath).forEach((keyName) => {
+        const contents = fs.readFileSync(`${this.config.publicKeysPath}/${keyName}`).toString();
+        this.publicKeys.set(keyName, contents);
+      });
+    }
+  }
+
+  private signObject(object: any, key: string): string {
+    const sign = crypto.createSign(this.SIGNATURE_TYPE);
```

Comment:
@idandash @netoneko it's hard for me to decide if it's important or not, but we actually shouldn't use RSA, since it's super hard to distribute RSA keys securely (regardless of future threshold encryption or not). The modern approach is to use a DLP (Discrete Logarithm Problem) based encryption (such as ECC) and then distributing key shares is basically a distribution of a scalar. 

Thinking even more about - it's not important for now. We should be able to easily change it in the near future.